donut/modules/uploads/templates/uploaded_list.html

```diff
@@ -0,0 +1,33 @@
+{% extends "layout.html" %}
+{% block scripts %}
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+  {{super()}}
+  <script src="https://cdn.rawgit.com/showdownjs/showdown/1.0.1/dist/showdown.min.js"></script>
+  <script src={{ url_for('uploads.static', filename='script.js', _external = True)}} type="text/javascript"></script>
+{% endblock %}
+{% block page %}
+<div class="container">
+  <div class="col-md-24">
+    <div class="jumbotron">
+      <h1 id = "title" class="text-center">List of Uploaded Pages</h1>
+      <div id = "content" style="font-size:21px;font-weight:200;
+        margin-bottom:15px;">
+        <table class = "table">
+          <tr>
+            <th>File name</th>
+            <th> </th>
+          </tr>
+          {% for link, name in links %}
+            <tr>
+              <td><a href= "{{ link }}">{{name}}</a></td>
+              {% if permissions%}
+                <td><a href= "{{ url_for('uploads.uploaded_list', filename=name)}}">Delete</td>
```

Comment:
The function uploaded_list looks for a filename from flask.request.args.get. If there are something, it deletes it.

Reply:
Hmm, that doesn't seem very intuitive. It might be clearer if you had a separate endpoint that did the deletion and then redirected the client to `uploads.uploaded_list`.